Adds simple demo script

mchurichi opened the pull request:
<!--
1. Please remember to include a DCO on every commit (`git commit -s`)
https://github.com/apps/dco
-->

**Pull Request check list**

- [ ] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

**Description of change**
Adds a set of simple demo scripts to demostrate how to setup federation between two SPIRE Server using Galadriel.

There is a Dockerfile and some make targets that are unrelated to the demo but might be useful.

**Which issue this PR fixes**
<!-- optional. `fixes #<issue number>` format will close an issue when this PR is merged -->

mgbcaio reviewed (approved):
Played around a little bit and it looks awesome! Fantastic work, Maxi.

Victorblsilveira commented on demo/README.md:
```diff
@@ -0,0 +1,30 @@
+
+![Demo diagram](diagram.png "Demo diagram")
+
+## Requirements
+
+This demo only requires a `git` client and the `make` command properly installed in the system.
+
```
There are any know restrictions related to environments like OS?

mchurichi replied:
That's a good question. I only tested on macOS, but as this is using standard bash it should also run on any Linux flavor and WSL, as long they have a relatively new version of bash. You are on Linux, right?

mgbcaio replied:
I only tested in macOS as well.

Victorblsilveira replied:
Does worth it to specify the environment in which we test it?

Victorblsilveira reviewed (dismissed):
Awesome! Just a Quick doubt raised.
LGTM.

mchurichi commented:
- Added a small note about what environments were used for testing.
- Added the ability to specify the SPIRE version/branch/tag we want to built. The default is the latest (v1.4.4) but it's currently [broken because of a dependency](https://github.com/spiffe/spire/issues/3506). In the meantime the main branch can be used with `SPIRE_VERSION=main ./0-setup.sh`

Victorblsilveira reviewed (dismissed):
LGTM

mgbcaio reviewed (approved):
🦖

jufantozzi reviewed (approved):
🥇

mchurichi commented:
Sorry folks, the automated branch update from github wasn't a signed commit so I had to squash it 🤦 I'll disable that feature to avoid the same issue in the future.

mchurichi commented:
god 🤦 this is the good one

mgbcaio reviewed (approved):
I got'chu

jufantozzi reviewed (approved):
lul
